### DISCUSSION TITLE
Placeholder for crowdin's language code?

### DISCUSSION THREAD

**Discussion by henieek** [crowdin/github-action#157] [Q&A] [answered]
Hello,

is there a placeholder code I could use to replicate the "crowdin code" documented here: https://developer.crowdin.com/language-codes/?

This format would work best for our usecase, however I'm unable to replicate it using placeholders I found here: https://developer.crowdin.com/configuration-file/#placeholders

What I want to achieve: before automating the upload process, the translation path for all source files seemed to have a default value of `%original_file_name%`. This caused the exported files to land under what seemed like `{crowdin_language_code}/%original_file_name%`.

However, when I try to configure my action like this:

```
  - source: app/src/main/res/values/*.xml
    translation: '%original_file_name%'
    translate_attributes: 0
```

the action complains about the translation path not having any country code:

```
'Translation' parameter should contain at least one language placeholder (e.g. %locale%)
```

I'm still however able to set these manually in the UI:

![image](https://user-images.githubusercontent.com/336158/211369789-821e51f8-41b5-4c67-9102-fed310e79075.png)

it would be great if we could keep the exported paths exactly as they were before we migrated to the Github Action - is that possible?

Thanks!

**Comment by henieek** [comment, marked as answer]
I figured out there's no placeholder, but I achieved what we needed by adding custom Language Mappings as described [here](https://support.crowdin.com/advanced-project-setup/). Thanks!

**Reply by paulgessinger** [reply]
So with these mappings, did you manage to make your translated files end up in the same location as the source files? This is what I'm trying to accomplish, but I'm not sure I can do that with language mappings.

**Reply by andrii-bodnar** [reply]
@paulgessinger I can help you with the setup. Could you please provide more details?

**Reply by paulgessinger** [reply]
Actually, I figured out that if i set `multilingual: true`, it let's me skip the locale replacement in the output translated file. I found this somewhere in the documentation, but I can't find it now anymore. (The documentation is a bit confusing for me).

Anyway, working for me now.

**Reply by andrii-bodnar** [reply]
I think it was here - https://crowdin.github.io/crowdin-cli/advanced#multilingual-files
